sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/GroupServiceImpl.java

```diff
@@ -205,98 +186,85 @@ public void update(String realm, String appName, Group group) {
               Map.entry(EventKeysConfig.GROUP, group),
               Map.entry(EventKeysConfig.APPLICATION_NAME, appName),
               Map.entry(EventKeysConfig.ERROR, e.toString())));
-      if (e instanceof GroupNotFoundException) {
-        throw (GroupNotFoundException) e;
-      } else {
-        throw e;
-      }
+      throw e;
     }
   }
 
+  // TODO attention il manque le userStorage du userId a ajouter
```

Comment:
can't the user be looked for in all userstorages ?

Reply:
yeah tje both are done, if storage is null user is looked in all us, but if we are  in asynchronous mode, user could not exist. So in synchronous ways userstorage can be null and we can search in all us, but in asynchronous ways user have no garantee that the user will be find and userstorage auto-completed